### PR TITLE
 Nimble: Added blecsc (Cycling Speed and Cadence) profile support for nimble examples. (IDFGH-10233)

### PR DESCRIPTION
**Pull request by ESPAbhinav**

### CODE REVIEW

**Comment by rahult-github** [comment]
sha=7dd75bbd733b35e4bb0258ecd9179444e8ed9b4b

**Comment by rahult-github** [comment]
Also, please change the copyright header in all files

File: examples/bluetooth/nimble/blecsc/README.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+
+It uses ESP32's Bluetooth controller and NimBLE stack based BLE host.
+
+This example aims at understanding CSC measurments, GATT database configuration, handling GATT reads and writes, handling subscribe events and understanding advertisement.
+
+
+Note :
+
+* To install the dependency packages needed, please refer to the top level [README file](../../../README.md#running-test-python-script-ttfw).
```
Comment (IshaESP) [inline examples/bluetooth/nimble/blecsc/README.md:26]: Remove the python script description. We don't have one for this example

File: examples/bluetooth/nimble/blecsc/main/blecsc_sens.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (IshaESP) [inline examples/bluetooth/nimble/blecsc/main/blecsc_sens.h:2]: SPDX header

File: examples/bluetooth/nimble/blecsc/main/gatt_svr.c [hunk 3]
```diff
@@ -0,0 +1,370 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (IshaESP) [inline examples/bluetooth/nimble/blecsc/main/gatt_svr.c:2]: Change headers of all files

File: examples/bluetooth/nimble/blecsc/main/gatt_svr.c [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+void
+gatt_svr_set_cp_indicate(uint8_t indication_status)
+{
+  csc_cp_indication_status = indication_status;
+}
+
+void
+gatt_svr_register_cb(struct ble_gatt_register_ctxt *ctxt, void *arg)
+{
+	char buf[BLE_UUID_STR_LEN];
```
Comment (IshaESP) [inline examples/bluetooth/nimble/blecsc/main/gatt_svr.c:333]: indentation

File: examples/bluetooth/nimble/blecsc/main/main.c [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+#include "blecsc_sens.h"
+#include "nimble/nimble_port.h"
+#include "nimble/nimble_port_freertos.h"
+
+#if CONFIG_EXAMPLE_EXTENDED_ADV
+static uint8_t ext_adv_pattern_1[] = { 
+    0x02, 0x01, 0x06,
+    0x03, 0x03, 0xab, 0xcd,
+    0x03, 0x03, 0x18, 0x11,
+    0x11, 0X09, 'n', 'i', 'm', 'b', 'l', 'e', '-', 'b', 'l', 'e', 'p', 'r', 'p', 'h', '-', 'e',
```
Comment (IshaESP) [inline examples/bluetooth/nimble/blecsc/main/main.c:37]: name should be nimble-blecsc-e

Comment (IshaESP) [reply examples/bluetooth/nimble/blecsc/main/main.c:37]: Update the length as well

File: examples/bluetooth/nimble/blecsc/main/main.c [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+ * then starts serving events from default event queue.
+ *
+ * @return int NOTE: this function should never return!
+ */
+int
+app_main(void)
+{
+    int rc;
+
+	 /* Initialize NVS — it is used to store PHY calibration data */
```
Comment (IshaESP) [inline examples/bluetooth/nimble/blecsc/main/main.c:304]: Indentation

**Comment by IshaESP** [comment]
Edit the examples/bluetooth/.build-test-rules.yml file as well

**Comment by rahult-github** [comment]
PR part of IDF. Closing

**Comment by Alvin1Zhang** [comment]
Thanks for contribution again, changes have been merged with https://github.com/espressif/esp-idf/commit/23bae9007043d84adc9b6e2813307a0937f336d0.

**Pull request closed without merging**